tower/src/ready_cache/cache.rs

```diff
@@ -367,6 +367,35 @@ where
 
         fut
     }
+
+    /// Call `Service::disarm` on all ready services and return them to the not-ready set.
+    ///
+    /// Returns the number of services disarmed this way.
+    ///
+    /// Note that `ReadyCache` eagerly polls services (one call to `poll_ready` may ready several
+    /// services), so a call to disarm may disarm a large number of underlying services.
+    pub fn disarm(&mut self) -> usize {
+        let were_ready = self.ready.len();
+
+        // Take self.ready so we can call &mut self methods below
+        let mut ready = std::mem::take(&mut self.ready);
+        for (key, (mut svc, cancel)) in ready.drain(..) {
```

Comment:
ouch :(

Reply:
Yeah... It's not _expensive_ (like, it won't allocate for example), but it does mean that you might un-ready a bunch of services.